src/compiler/checker.ts

```diff
@@ -29501,6 +29501,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return !isPastLastAssignment(symbol, /*location*/ undefined);
     }
 
+    // Check if a parameter or catch variable is assigned definitely
+    function isSymbolAssignedDefinitely(symbol: Symbol) {
+        return symbol.isDefinitelyAssigned || (isSymbolAssigned(symbol) && symbol.isDefinitelyAssigned);

```

Comment:
Isn't the right thing here

```suggestion
        return symbol.isDefinitelyAssigned ?? (isSymbolAssigned(symbol) && symbol.isDefinitelyAssigned);
```

Otherwise, you will recalculate `isSymbolAssigned` every time you want to run `isSymbolDefinitelyAssigned` in the case where a variable is never assigned to.

Reply:
fixed